File: iherefor-smart-development/demo/reader-app/android/app/src/main/java/com/iherefor/readerapp/parsing/BookParser.kt
package com.iherefor.readerapp.parsing

import com.iherefor.readerapp.domain.ParsedBook

/** 可替换解析 seam（对齐 跨端共享层契约，未来可换 Readium） */
interface BookParser {
    fun parse(data: ByteArray): ParsedBook
}

class ParserException(message: String) : Exception(message)

File: iherefor-smart-development/demo/reader-app/ios/ReaderApp/Parsing/BookParser.swift
import Foundation

/// 可替换解析 seam（对齐 跨端共享层契约，未来可换 Readium）
protocol BookParser {
    /// 从字节流解析出文档流（章节序列）。失败抛错。
    func parse(data: Data) throws -> ParsedBook
}
